interceptors/interceptors.go

```diff
@@ -137,6 +138,27 @@ func ServerErrorInterceptor() grpc.UnaryServerInterceptor {
 	}
 }
 
+func PanicRecoveryInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+		defer func(ctx context.Context) {
+			// panic handler
+			if r := recover(); r != nil {
+				log.Error(ctx, "panic", r, "method", info.FullMethod)
+				if e, ok := r.(error); ok {
+					err = e
+				} else {
+					err = errors.New(fmt.Sprintf("panic: %s", r))
+				}
```

Comment:
what is the benefit of this condition check as opposed to just doing `err = errors.New(fmt.Sprintf("panic: %s", r))`?

Reply:
good question, I'm not sure as I just copied it to ensure we use the same logic.